telink: tl721x: update CLI/RCP demo for Tercel A1.

pulin1103 opened the pull request:
- update conf file.
- enable CONFIG_IEEE802154_TLX_MAC_FLASH for TLX chips
- enable CI cli/rcp_usb demo.

liucoldstar commented on .github/workflows/telink-tl721x-build.yaml:
```diff
@@ -85,25 +85,15 @@ jobs:
-    #     cd ..
-    #     west build -b tl7218x                  -d build_ot_coprocessor_rcp_uart_tl721x   zephyr/samples/net/openthread/coprocessor                -- -DCONFIG_COMPILER_WARNINGS_AS_ERRORS=y -DOVERLAY_CONFIG=overlay-rcp.conf
-
-    # - name: Build TL721X net/sockets/echo_server for OpenThread
```
echo_server和echo_client能编译过吗？如果可以的话，把这两个例程的编译也使能。

pulin1103 replied:
已添加

liucoldstar commented on samples/net/openthread/coprocessor/boards/tl7218x-dongle.overlay:
```diff
@@ -0,0 +1,31 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/pinctrl/b95-pinctrl.h>
```
移除所有对B5的引用

pulin1103 replied:
已修改

liucoldstar commented on samples/net/openthread/coprocessor/boards/tl7218x-dongle.overlay:
```diff
@@ -0,0 +1,31 @@
+&pinctrl {
+	pad-mul-sel = <1>;
+
+	uart0_tx_pa3_default: uart0_tx_pa3_default {
```
不要有对B9x的引用

pulin1103 replied:
已修改

liucoldstar commented on samples/net/openthread/cli/boards/tl3218x.conf:
```diff
@@ -28,3 +28,4 @@ CONFIG_OPENTHREAD_COAPS=n
 CONFIG_OPENTHREAD_ENABLE_SERVICE=n
 CONFIG_OPENTHREAD_CSL_RECEIVE_TIME_AHEAD=480
 CONFIG_OPENTHREAD_MAX_CHILDREN=5
```
对TL3218x的所有修改放到一个独立的PR里，直接向develop和develop_new分支提。

pulin1103 replied:
这是修改之前的TLX的引用错误，应该放在这个PR里，先合到tl721x中，然后再合到develop

liucoldstar replied:
@pulin1103 TL3218x已经合入了develop分支，已经是发布状态，任何TL3218x都直接向develop和develop_new分支提交，保证及时修复bug。

pulin1103 replied:
已修改

Commit:
telink: tl721x: update CLI/RCP demo for Tercel A1.

- update conf file.
- enable CONFIG_IEEE802154_TLX_MAC_FLASH for TLX chips
- enable CI cli/rcp_usb echo_client/server demo.

Signed-off-by: Zhenghuan Zhang <zhenghuan.zhang@telink-semi.com>
